app/controllers/twitch_stream_updates_controller.rb

```diff
@@ -0,0 +1,40 @@
+class TwitchStreamUpdatesController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def show
+    if params["hub.mode"] == "denied"
+      Rails.logger.error("Twitch Webhook was denied: #{params.to_json}")
+      head :no_content
+    else
+      render plain: params["hub.challenge"]
+    end
+  end
+
+  def create
+    head :no_content
+
+    unless secret_verified?
+      Rails.logger.warn("Twitch Webhook Recieved for which the webhook could not be verified")
+      return
```

Comment:
We return a `204` (due to https://github.com/thepracticaldev/dev.to/pull/2591/files#diff-0412fe5589089c679dcfde2206fc9a2eR14) from this even if we fail to verify the secret. This is because if someone is trying to fake the webhook requests, there is no reason to tell the attacker something failed. Better to simply silently noop and not tell them anything is amiss 

We could also choose to raise an error here during the rollout until we are more certain this is integration works and there is no issues with verifying secrets with Twitch